More features from SRPG_GearMZ

RyanBram opened the pull request:
Dear Mr 有明タクミ,

The following GitHub commit provides two changes:

1. Window_ActorCommand.prototype.updatePlacement   Added an offset to adapt to the differences between `Graphics.boxWidth` and `Graphics.width`.

2. Added Touch UI similar to RPG Maker MZ.

Both require Community_Basic.js. I kindly ask you to include it in the distribution. Community_Basic.js is useful for aligning RPG Maker MZ features to be used in MV.

---
有明タクミ様

以下のGitHubコミットには2つの変更が含まれています：

1. Window_ActorCommand.prototype.updatePlacement   `Graphics.boxWidth` と `Graphics.width` の違いに適応するためにオフセットを追加しました。

2. RPG Maker MZと同様のタッチUIを追加しました。

どちらもCommunity_Basic.jsが必要です。配布に含めていただけますようお願い申し上げます。Community_Basic.jsは、RPG Maker MZの機能をMVで使用できるように調整するために役立ちます。

RyanBram committed:
More features from SRPG_GearMZ

Dear Mr 有明タクミ,

The following GitHub commit provides two changes:

1. Window_ActorCommand.prototype.updatePlacement  
   Added an offset to adapt to the differences between `Graphics.boxWidth` and `Graphics.width`.

2. Added Touch UI similar to RPG Maker MZ.

Both require Community_Basic.js. I kindly ask you to include it in the distribution. Community_Basic.js is useful for aligning RPG Maker MZ features to be used in MV.

---
有明タクミ様

以下のGitHubコミットには2つの変更が含まれています：

1. Window_ActorCommand.prototype.updatePlacement  
   `Graphics.boxWidth` と `Graphics.width` の違いに適応するためにオフセットを追加しました。

2. RPG Maker MZと同様のタッチUIを追加しました。

どちらもCommunity_Basic.jsが必要です。配布に含めていただけますようお願い申し上げます。Community_Basic.jsは、RPG Maker MZの機能をMVで使用できるように調整するために役立ちます。

RyanBram commented:
Attached is the Project File.
プロジェクトファイルを添付しました。
[https://drive.google.com/file/d/1e61qJa29aokOGiCKxRZbk2zH3Eooa5Gv/view?usp=sharing](url)

RyanBram commented:
The Touch UI still has a bug where pressing the cancel button during an SRPG Battle opens the menu instead of canceling the previous action.

---
タッチUIにはまだバグがあります。SRPGバトル中にキャンセルボタンを押すと、前の操作をキャンセルする代わりにメニューが開いてしまいます。

RyanBram commented:
After updating Community_Basic.js, I have successfully eliminated the bugs in the “Menu Button” and “Cancel Button”.

I think it should be stable enough to use. Please download the Google Drive file that I have updated at the link above.

---

Community_Basic.jsを更新した結果、「メニューボタン」と「キャンセルボタン」のバグを取り除くことに成功しました。

これで安定して使えると思う。更新したGoogle Driveのファイルは上のリンクからダウンロードしてください。

Ohisama-Craft commented:
Thank you, Ryan!
I think the reverse porting of the touch UI from MZ to MV is a very interesting endeavor.

I’ve reviewed the project files.
I have some concerns about incorporating this feature that I’d like to share.

Handling of Button Images
The project file you sent uses MZ button images. Using these images could raise legal issues, such as requiring MZ user registration for usage rights. It’s necessary to provide original images.

Handling of community_Basic
This plugin is automatically inserted when creating a new project. Directly modifying and redistributing such a commonly used plugin might confuse users. Which parts of community_Basic.js were added by you?

Compatibility with Other Plugins
It’s essential to ensure compatibility with plugins like SRPG_MouseOperation and SRPG_TerrainEffectPlus. For example, preventing the screen from scrolling when the mouse is over a button or ensuring the buttons don’t overlap with the terrain effect window.
Additionally, if the screen size matches the UI area size, the buttons may overlap with menu and help windows.

I assume this plugin is designed for environments where the screen size is larger than the UI area size. This would resolve many of the issues mentioned in point 3. It might be helpful to mention this in the plugin help section.
In such an environment, this plugin would allow the MZ UI to be used effectively in MV.

Would it be possible to consolidate the changes to SRPG_core and community_Basic into a single, standalone plugin? Since I’m not sure which parts of community_Basic are your original contributions, I can’t fully assess the feasibility.
Another task to consider is creating original button images.

Please consider these points.
Thank you so much!

ありがとう、Ryan！
タッチUIのMZからMVへの逆移植は、とても面白い試みだと思います。

プロジェクトファイルも確認しました。
この機能を組み込むうえで、いくつか懸念がありますのでお伝えさせてください。

１．ボタン用の画像の扱い
送ってもらったプロジェクトファイルではMZのボタンの画像が使用されていましたが、このような形にすると使用に際してMZのユーザー登録も必要になってくるなど権利関係の問題が生じます。
オリジナルの画像を用意する必要があります。

２．community_Basicの扱い
このプラグインは、プロジェクトを新規作成した時に自動で挿入されるプラグインだったと思います。
多くのゲームで共通して用いられるプラグインを直接書き換えて再配布するのは、ユーザーを混乱させるのではと心配しています。
このcommunity_Basic.jsのどの部分までが、あなたが追加してくれた部分ですか？

３．他のプラグインとの整合性
SRPG_MouseOperationやSRPG_TerrainEffectPlusと整合性を取る必要があります。
たとえば、ボタンにマウスを合わせた時に画面がスクロールしないようにする、ボタンが地形効果ウィンドウと重ならないようにする、など…
また、画面サイズとUI areaのサイズを同じにすると、メニューウィンドウやヘルプウィンドウにボタンが重なってしまいます。


察するに、これは画面サイズがUI area サイズより大きいことを前提としたプラグインだと思います（そうすると、“３．他のプラグインとの整合性”で挙げた問題の多くは解決します）。そのことはプラグインヘルプなどにも書いておいた方が良いかもしれません。
そのような環境においては、MZのUIをMVに逆輸入して使える良いプラグインだと思います。

SRPG_coreとcommunity_Basicへの変更をまとめて、独立した一つのプラグインにすることは出来ませんか？　（community_Basicのどの部分までがRyanのオリジナルか分からないので、可能か判断しきれないのですが…）
また、解決すべき課題の一つはオリジナルのボタン画像を用意することだと思います。

検討してみてください。
Thank you so much!

RyanBram commented:
Hi, Mr. 有明タクミ.
Thank you very much for your valuable feedback and input.

Among the reasons I updated Community_Basic.js was when I realized that certain features existed in RPG Maker MV, but because the implementation was still buggy, I didn't see them being used much. For example, Graphics.width and Graphics.boxWidth have been around since RPG Maker MV launched, but it wasn't until RPG Maker MZ that I often saw that they could be used separately, so that a game could better adapt to multiple screen ratios, because only of this small mistake:
![Community Difference](https://github.com/user-attachments/assets/6064c8be-330c-4e63-95e7-90dfa7b2437e)

Even with my simple fixes, the following bugs can still be found:
![Graphic Bug](https://github.com/user-attachments/assets/807dda42-406a-4f1f-876a-30de781b4793)


Just because of this very small mistakes in RPG_Core.js:
![Mask Issue](https://github.com/user-attachments/assets/76e43eb7-4bb0-4342-a898-1ccf2637ef9e)


Since RPG Maker MV is almost impossible to be officially updated, I can only update the RPG_Core.js through plugins if I want to get features that should have been there in the first place.

Regarding your concern, I tried to do the following:

1. Make my own revisions to Community_Basic.js only for the required features separately. I even tried to keep it as simple as the original.
Attachment: [Community_Basic.js.txt](https://github.com/user-attachments/files/18335291/Community_Basic.js.txt)

2. Made the additions that I backported and customized from RPG Maker MZ as a separate plugin.
Attachment: 
[TouchUI.js.txt](https://github.com/user-attachments/files/18335315/TouchUI.js.txt)

3. Preparing touch button set based on modification of RPG Maker MV buttons set.
Attachment: 
![TouchButtonSet](https://github.com/user-attachments/assets/4627be2e-c797-465a-a60e-2fec7e949e85)

4. Regarding the pull request I added on GitHub, it was an addition that I put in SRPG_core.js and it was actually just a backport of your own code in SRPG_core_MZ.js.

Thank you very much for your help. I hope that in 2025 I can again take the time to help you in developing this plugin that we both love.

Ohisama-Craft commented:
Thank you for the detailed response, Ryan!
I appreciate your pointing out bugs that I wasn’t aware of.
I’m also grateful for the original images you provided.

Let me confirm the final steps for implementation:
1. Distribute Community_Basic.js and have users overwrite their existing Community_Basic.js.
2. Apply the changes made in the pull request to SRPG_core.js.
3. Have users install TouchUI.js.
4. The images are modified versions of the button images from RPG Maker MV.

Is everything correct with the above steps?
Thank you, and I look forward to working with you this year as well!
